tools/xcodeproj_shims/installers/_swiftmodule.sh

```diff
@@ -2,18 +2,31 @@
 
 set -euo pipefail
 
-for module in "$@"; do
-    doc="${module%.swiftmodule}.swiftdoc"
-    module_name=$(basename "$module")
-    module_bundle="$BUILT_PRODUCTS_DIR/$module_name"
-    mkdir -p "$module_bundle"
+# Xcode 12 and under stored index information in /DerivedData/<project>/Build/Products
+readonly xcode_12_index_dir="$BUILT_PRODUCTS_DIR"
 
-    cp "$module" "$module_bundle/$CURRENT_ARCH.swiftmodule"
-    cp "$doc" "$module_bundle/$CURRENT_ARCH.swiftdoc"
+# Xcode 13 and later store index information in /DerivedData/<project>/Index/Build/Products
+readonly xcode_13_index_dir="${BUILT_PRODUCTS_DIR/\/Build\/Products\///Index/Build/Products/}"
 
-    ios_module_name="$CURRENT_ARCH-$LLVM_TARGET_TRIPLE_VENDOR-$SWIFT_PLATFORM_TARGET_PREFIX$LLVM_TARGET_TRIPLE_SUFFIX"
-    cp "$module" "$module_bundle/$ios_module_name.swiftmodule"
-    cp "$doc" "$module_bundle/$ios_module_name.swiftdoc"
+readonly dirs=(
+  "$xcode_12_index_dir"
+  "$xcode_13_index_dir"
+)
 
-    chmod -R +w "$module_bundle"
+for index_dir in "${dirs[@]}"; do
```

Comment:
Will this impact build times? Should we find a way to only copy to the Xcode being used? Not sure just curious to know what others think 🤔

Reply:
This is executed in another process, so it will not delay build times.

I think Xcode 13 _may_ still use both locations, so my variable naming might not be semantically correct, but I do think it is a good idea to keep copying to both locations.

I asked this question today as well: https://ios-dev-at-scale.slack.com/archives/CJDFFEN4S/p1632835902014300?thread_ts=1631798553.000500&cid=CJDFFEN4S

Reply:
Awesome thx for asking that question and clarifying it here 🙏!